src/main/scala/com/evolutiongaming/retry/Retry.scala

```diff
@@ -25,10 +26,10 @@ object Retry {
   }
 
 
-  def apply[F[_], E](
+  def apply[F[_] : Sleep, E](
```

Comment:
not backward compatible changes

Reply:
@t3hnar It is source compatible because `GenTemporal` will get converted to `Sleep` implicitly... It is not binary compatible though.

I can make it binary compatible by reworking these methods to use `GenTemporal` instead and make `Sleep` logic private or even get rid of `Clock` constraint either. :thinking:

Reply:
Like this:
```scala
  def apply[F[_], E](
    strategy: Strategy)(implicit
    F: GenTemporal[F, E]
  ): Retry[F] = apply(strategy, OnError.empty[F, E])


  def apply[F[_], E](
    strategy: Strategy,
    onError: OnError[F, E])(implicit
    F: GenTemporal[F, E]
  ): Retry[F] = fromSleep(strategy, onError)


  private[retry] def fromSleep[F[_] : Sleep, E](
    strategy: Strategy,
    onError: OnError[F, E])(implicit
    F: MonadError[F, E]
  ): Retry[F] = { ... implementation goes here ... }
```

Reply:
@Z1kkurat your word here, how many libs did you update with latest `retry` release?

Reply:
I haven't yet updated any libs with it AFAIR